tasks/provision_newstyle_compiler.json

```diff
@@ -0,0 +1,12 @@
+{
+  "puppet_task_version": 1,
+  "supports_noop": false,
+  "description": "Provision New Style Compiler(s)",
+  "parameters": {
+    "compiler": {
```

Comment:
If this supports multiple compilers, it should be plural and accept denote the format of the input when sending multiple compilers in.